### PR TITLE
Refactor

### PR DESCRIPTION
**Pull request by AkashaRojee**
Hello!

We refactored the project.

- `main.rb` now only launches app, shows menu and handles user input.
- `app.rb` then processes the menu option selected by user.

Please, review the project.

Thanks!

### CODE REVIEW

File: app.rb [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+    parent_permission = gets.chomp != 'n'
+
+    @persons.push(Student.new(age, 'learn to code', name, parent_permission))
+  end
+
+  def create_person
+    print 'Do you want to create a student (1) or a teacher (2)? [Input the number]: '
+    person_type = 0
+
+    person_type = gets.chomp while person_type != '2' && person_type != '1'
```
Comment (AshakaE) [inline app.rb:61]: All requests for input and input validation should be placed in the `main.rb` file

- [ ] Kindly refactor code

File: app.rb [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+
+    case person_type
+    when '1'
+      create_student
+    when '2'
+      create_teacher
+    end
+
+    puts 'Person created successfully'
+  end
```
Comment (AshakaE) [inline app.rb:71]: Try to follow the [best practices and solid]( https://github.com/microverseinc/curriculum-ruby/blob/main/oop/lessons/solid.md#best-practices-and-solid) 

- [ ] Kindly refactor code

File: app.rb [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+    end
+    selected_person = gets.chomp.to_i
+    person = @persons[selected_person]
+
+    print 'Date: '
+    date = gets.chomp
+
+    @rentals.push(Rental.new(date, person, book))
+    puts 'Rental created successfully'
+  end
```
Comment (AshakaE) [inline app.rb:104]: Try to follow the [best practices and solid]( https://github.com/microverseinc/curriculum-ruby/blob/main/oop/lessons/solid.md#best-practices-and-solid) 

- [ ] Kindly refactor code

**Review by AshakaE — CHANGES_REQUESTED** [review]
### Hi @AkashaRojee & @moise10r  🙋‍♀️, 

Good job so far! 👌
Pull request 👌 
Git flow ✔️ 
Readme  👍 
Your abstraction with the `app.rb` is great but it needs more work 💡 

There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>


#  Required Changes ♻️ 🚧
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by SpaYco — APPROVED** [review]
Hello 👋,

_This is an additional review as a result of the second opinion request._

I'm sorry this took longer than it should, it totally passed by and got lost in the queue,

You're right, the reasons sent by the CR is for 1.0 projects, so you're good to go here

As this was the only reason that your project was not approved it will be approved as soon as possible.
Please ask for another review in your Dashboard and the next Code Reviewer will mark it as approve right away :)

Keep rocking 🚀!

**Review by emmanuellekamwa — APPROVED** [review]
**STATUS: APPROVED 🎉✔**
Hello @AkashaRojee 👋🏽
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**